### ISSUE TITLE
Measure the decision window (settle 10 and 20) as the next arm toward model-only 1-1 completion

### ISSUE THREAD

**Issue by EthanThatOneKid** [EthanThatOneKid/super-fly#36]
## Goal

Hand this project off cleanly at the point it has actually reached, and name the next experiment.
#30 is not solved and the reason is now measured rather than guessed: the decision the controller
makes each chunk cannot currently be made well enough to chain twenty jumps, and the offline tools
that say so are new, tested, and reusable.

The next experiment is the **decision window** (`--settle-steps 10` and `20`), with two gate
defects fixed first because they are cheap and they affect how any result is read.

## Where this stands

**The gate-eligible closed-loop result, unchanged.** Model-only on the reserved seeds, real ROM,
`model_only: true`, `p0_gate_met: false`: untrained `best_x` 898, DAgger round 1 **1247**, round 2
899, completion `0.0` in every arm. The teacher plan completes the level in 1,477 frames
(`max_x` 3243, no death), so the upper bound and the completion detector are real. Two decoder
defects that blocked the gate are fixed and are worth 313 -> 1247 on their own: the jump channel
was cancelling itself out of the evidence sum, and the jump margin was hard-coded to zero on a
readout whose evidence is offset for both chunk classes.

**The measurement side is now honest, and that took three corrections.** All iteration moved to the
dev seeds `45-49` and the reserved `42/43/44` became report-only, because every offline verdict had
been read off the seeds the claim has to be made on (`prescreen.py` and `pretrain.py` now *refuse*
the reserved set). The pre-screen became a teacher-forced sequential replay with a rate budget,
after the chunk-at-a-time versions were either unfalsifiable (a boundary fitted on the evidence it
then scored) or gameable (an always-jump policy scored a perfect recall and a completion). And the
paired criterion compares every arm against an untrained baseline measured on the *same* replays,
because the baseline's own recall moves with the seed set (0.733 on the reserved three, 0.800 on the
dev five, same shard and metric).

**What the arms do today**, dev seeds, five replays: round 1 takes 0.840 of the teacher's jumps in
sequence, round 2 0.810, untrained 0.800; the budget view reads 0.496 / 0.504 / 0.360 against 0.343
chance. Nothing passes, and the failure is always the same two criteria: completion, and a paired
gain over the initialization (+0.040 and +0.010 against a +0.10 floor).

**The ceiling is priced.** `oracle.py` scores every settle-window statistic on identical draws with
the threshold *and* the timing chosen from the labels, at bounded jump cost, and probes the
central-complex population one layer upstream from the same forward passes. At the shipped window
the evidence's own ceiling is 0.656 bounded per-jump recall where the shipped statistic achieves
0.056, and it climbs to 0.928-0.952 (AUC 0.99) when a decision is allowed twenty draws. The
population upstream does not rescue it (held-out readouts reach 0.448 at settle 5 and 0.168 at 20,
below the motor ceiling; the same readout fitted in-sample scores 1.000, which is a fit). The
central population is also **bit-identical** across the untrained baseline and both DAgger rounds --
they differ only in `layer3_4`, 512 weights over 128 central units -- so "we trained the network" has
so far meant "we trained the readout".

**The cheap decoder margin was taken, and it did not pay.** The settle-window statistic is now a
parameter (`evidence.py`): eight rules including the shipped spike sum, the rule and its decay
carried in the checkpoint, the margin re-fitted with its provenance recorded. Forced across every
arm on the dev seeds, five statistics: the rate-free budget view barely moves (untrained
0.336-0.360, round 1 0.480-0.544), while the sequential view swings by 0.30 on every arm and tracks
the jump rate a calibrated margin spends -- the untrained arm reaches 0.960 twice, at rates 0.374
and 0.404, both over the 0.35 budget, where at its own budget its recall is 0.336. `spike_sum` stays
the default; the drive rules stay reachable per arm. The ceiling's resolution finding was real but
confined to the ceiling's own metric.

**Two things now make the sequential view hard to trust as a ranking.** The round-1/round-2 ordering
flips with the seed set (round 2 ahead on the reserved three, round 1 ahead on the dev five, against
an emulator that has round 1 ahead). It also flips with the statistic (`spike_leaky_recency` puts
round 2 ahead where the shipped sum puts round 1 ahead, same data, same seeds). It is still the gate
and it is what caught the always-jump exploit -- but it is not a ranker at n=5.

**Nothing is ruled out by a number that was read off the wrong seeds any more.** Ruled out, with
measurements: teacher-only pretraining, per-frame rather than per-cadence supervision, progress-only
recovery labelling (a real defect, worth 1247 -> 1246), visual-pathway training (a statistical
regression, and it saturates the readout until the margin cannot separate the classes), the decoder's
channel arithmetic and zero margin (fixed), and now the settle-window statistic.

**The harness exists.** `sweep.py` merges tables by provenance (shard checksum, protocol, seed role,
statistic per arm), diffs them against a reference, and exits non-zero when a published number moves.
`.github/workflows/prescreen-sweep.yml` runs a set of cells as a matrix on manual dispatch, one job
per cell, and calls the same merge in an aggregate job that writes the table into the run summary and
uploads a single artifact -- so a sweep is one dispatch and one download instead of a serial
afternoon. 295 tests pass; the branch is `issue-30-closed-loop-dagger` (commits above `6b1ad81`).

## Work

1. **Hold the baseline to its own budget.** The paired criterion compares an arm against an untrained
   baseline whose margin is calibrated the same way but whose *rate* is unconstrained, so with the
   flat drive the baseline spends 37.4% of its decisions jumping against a 35% budget and every
   trained arm "fails" `beats_untrained` against a bar it is not allowed to match. Fix the rate (or
   compare at a matched rate), with a test that a baseline cannot win the comparison by overspending.
2. **State what the gate may conclude.** Given that the sequential ordering is not stable across
   seeds or statistics, decide the rule explicitly: what a recall floor means at n=5, whether a
   per-replay spread larger than the paired gain should block a verdict, and which single view a
   pass must hold on. Write it next to the criteria, so a verdict is reproducible rather than
   argued.
3. **Measure the window.** Re-run the same instrument at `--settle-steps 10` and `20` on the dev
   seeds: ceiling, sequence pre-screen, budget view, against the same arms. This is offline, costs
   no emulator time, and is the only lever found so far that moves a ceiling near the 0.97
   completion needs.
4. **Then, and only then, spend an emulator run.** If an arm clears the pre-screen at the recall
   floor and inside the budget with a consistent paired gain, freeze it and evaluate model-only on
   the reserved seeds `42/43/44` -- one arm, three seeds, reported honestly as
   `claim_eligible` or not.
5. **Write the negative result if that is what it is.** If the window does not produce a candidate,
   the deliverable is the measurement plus a narrowed scope, not another sweep: the visual pathway
   is the remaining unknown (training it *was* worse than not training it, and the escape hatch is
   a surrogate-gradient path into it, which is genuinely new capability).

## Acceptance criteria

- The pre-screen's paired criterion cannot be won by a baseline that overspends the jump budget, and
  a test pins that.
- The gate's decision rule is written down, and every verdict in the report names the view it came
  from.
- Settle 10 and 20 are measured on the dev seeds with both views and the ceiling, jump rate
  reported, and the numbers recorded in `README.md` and `REACH_1_1_PLAN.md`.
- At most one closed-loop arm per window is spent, on the reserved seeds, with `model_only: true`
  and the completion detector unchanged (`x >= 3200`, state `0x05` for 30 frames).
- No claim is made from a dev-seed reading, and no arm is described as passing because a *baseline*
  did worse.

## Non-goals

Re-open the statistic (measured, no rate-free gain), keep sweeping the motor readout (<= 1 px and
negative so far), tune hyperparameters to chase the gate, change FlyWire/Hemibrain topology or visual
encoding, add WebGPU or distributed execution, or report a completion the model did not achieve.
Settle 20 is a *different controller* -- four times the inference cost and different adaptation --
so it is a new configuration to measure, not a setting to switch on.

## Open decision, blocking CI sweeps

The workflow needs the teacher shard and three checkpoints **on the runner**; it publishes nothing
itself. This is smaller than it first looked:

- The ROM is already tracked (`roms/Super Mario Bros. (World).nes`), and the teacher shard is
  generated from it deterministically, so a runner can regenerate the 3.2 MB shard and check its
  manifest checksum rather than being sent one.
- That leaves the three checkpoints (~12 MB, `runs/` is gitignored) as the only inputs with no
  source. Options: commit them to the repo, regenerate them on the runner (they need DAgger rounds),
  use a self-hosted runner that already holds `runs/`, or keep data-bearing sweeps local and use CI
  for the test suite only.
- Cost is not the constraint: the repository is public, so standard GitHub-hosted runners are
  unmetered.

Recommendation: regenerate the shard from the tracked ROM on the runner and commit the three
checkpoints (or a single "candidate" checkpoint that all four arms can be built from), so a sweep is
one checkout and one dispatch with nothing to prepare by hand. #34 owns the wider reproducibility
work this belongs to.

## Related work

- #30 -- closed-loop DAgger and macro-action experiment; this issue is where it has actually got to
- #31 -- reserved-seed P0 benchmark and first model-only checkpoint (the consumer of item 4)
- #34 -- reproducibility on a clean machine and in CI (owns the data question above)
- #3, #10 -- autonomous jump timing and offline trajectory pretraining
- Branch `issue-30-closed-loop-dagger`: `2d406e5` the ceiling instrument, `c444352` the statistic
  parameter and its rejection, `e397c5c` the sweep merge and the dispatch workflow, `e7dd09e` the write-up

**Comment by EthanThatOneKid** [comment]
### Dispatch evidence for the CI decision at the bottom

The branch is pushed (`d6046fc`) with the sweep workflow in it, and the first dispatch attempt
reports a prerequisite that is not obvious from the trigger documentation:

```
$ gh workflow run prescreen-sweep.yml --ref issue-30-closed-loop-dagger \
    -f cells='[{"id":"spike","rule":"spike_sum"},{"id":"drive","rule":"drive_sum"}]'
HTTP 404: workflow prescreen-sweep.yml not found on the default branch
```

The contrast explains it: `ci.yml` dispatches fine on the `ci-workflow` branch even though that
file is no longer on `main`, because it *was* on `main` once and the workflow is therefore
registered in the repository's Actions. A workflow file that has never been on the default branch
cannot be dispatched at all, by path or by name -- `--ref` selects which revision to *run*, not
which workflow exists.

So the prerequisite for the first dispatch is landing `.github/workflows/prescreen-sweep.yml` on
`main` (a PR is the natural route), after which any pushed ref can be swept. Everything below the
platform layer is already verified: the YAML parses, `sweep.py` merges and gates on real tables
(five statistics, 30 tests), and the cells' shell logic mirrors the local runs that produced the
numbers in the README. What is not yet verified is the matrix expansion and the aggregate step
against GitHub itself, which is what the first dispatch will settle.

Worth noting for whoever picks this up: with the file on `main`, a sweep of the *old* branches
still works, so this does not need a merge of the whole branch to unblock.

**Comment by EthanThatOneKid** [comment]
**The first reference dispatch landed, and it reproduced the published table exactly.** [Run 35933466810](https://github.com/EthanThatOneKid/super-fly/actions/runs/35933466810) (`ubuntu-latest`, py3.11, cell `ref-untrained`, forced `spike_sum@0.25`, settle 5 / stride 15 / dev 45-49, tolerance 0), total wall time ~9 min.

Two results, both of which the issue's work rested on:

- **The shard regenerates byte-identically on the runner.** `data/` is gitignored, so the cell regenerated the teacher shard from the tracked ROM with `teacher.py` and hashed `shard-00000.npz` to `c050a0a0…4290b8` — the reference's bytes, from a different numpy/zlib build than the one this table was made with. So the teacher trajectory is a property of the ROM, and the ROM is already reached by `actions/checkout`; the "the ROM is the blocker" framing in my earlier comment on this issue was wrong.
- **The numbers are host-independent at `--tolerance 0`.** `1 row(s) compared, 0 moved beyond 0.0`, with *every* field equal — per-replay sequence recall 0.70/0.65/0.70/0.70/0.55, jump rate 0.2121/0.1717/0.1717/0.202/0.202, spurious 7/4/3/6/9, `best_x` 549/362/249/249/249, and the re-fitted margin's calibration block (balanced accuracy 0.519). A 2-vCPU runner's float reduction order does not move a digit against the local eight-core run, so dispatched sweeps are comparable with local readings without a per-host caveat.

Two defects in the harness surfaced on the way, both of which made the first dispatch fail or produce nothing:

1. **Every cell was held to the same three checkpoints**, so the untrained baseline — the only row a runner without weights can reproduce — was blocked by data it never reads. Checkpoints are now declared per cell, with `none` as the explicit empty list (an empty string falls back to the input's default).
2. **No cell passed `--output`**, so no cell wrote a table and the merge had nothing to compare even when the cell ran. Each cell now writes `runs/prescreen/<id>.json`, and a cell that ran and wrote no table fails instead of warning.

The workflow also names a shard/reference mismatch before the run rather than reporting it as drift after it, since rows are matched on the shard's *file bytes* — a mismatch and a drifting float both surface as "nothing compared" from `sweep.py`. `tests/test_workflow.py` (7 tests, stdlib-only, each assertion fails against the previous workflow) now guards that wiring. Commits `7d061e3` and `76168f2` on `issue-30-closed-loop-dagger`; suite at 305.

**What is left for the gate to cover the whole table:** the three checkpoint-bearing rows need `runs/closed_loop_dagger/iter-01/checkpoint.pth`, `runs/closed_loop_dagger/round2/iter-01/checkpoint.pth` and `runs/prescreen/feedback.pth` on the runner — 13 MB, all under the gitignored `runs/`. Either they move to a tracked path (or a release artifact) and the gate covers all four rows, or CI regenerates them from the ROM, which is now a more plausible option than it looked: the emulator installs and runs headless on `ubuntu-latest`.

**Comment by EthanThatOneKid** [comment]
Filed the data-side of the open decision as its own deliverable: #38, *Get the three checkpoints onto a runner so the reference gate covers all four rows of the pre-screen table* (`priority:next`). It carries the state after the reference run above -- what is proven, the two harness defects that were fixed, why each checkpoint is load-bearing (they carry the margin and calibration the published rows were measured at), and the four transport options with a recommendation (a release asset, so no weights enter history and no secret is needed on a public repo). The ROM/shard half of the question in the *Open decision* section above is resolved by the run and needs no choice.
